sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/OrderedDistinctMap.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.Resource;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+public class OrderedDistinctMap extends DistinctMap {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);;
+    private String lastHash;
```

Comment:
is `lastHash` field value needed?
if so shouldn't this be volatile to ensure threads have consistent view on the value?

Reply:
yup, made this volatile